server/db/db.js

```diff
@@ -1175,6 +1179,281 @@ async function getDisconnectedGatewaysWithClient(client, pgClient) {
   return null
 }
 
+async function getButtonWithId(id, pgClient) {
+  try {
+    const results = await helpers.runQuery(
+      'getButtonWithId',
+      `
+      SELECT *
+      FROM buttons
+      WHERE id = $1
+      `,
+      [id],
+      pool,
+      pgClient,
+    )
+
+    if (results.rows.length > 0) {
+      const allClients = await getClients(pgClient)
+      return createButtonFromRow(results.rows[0], allClients)
+    }
+  } catch (err) {
+    helpers.logError(err.toString())
+  }
+
+  return null
+}
+
+async function getButtonsWithClientId(clientId, pgClient) {
+  try {
+    const results = await helpers.runQuery(
+      'getButtonsWithClientId',
+      `
+      SELECT *
+      FROM buttons
+      WHERE client_id = $1
+      `,
+      [clientId],
+      pool,
+      pgClient,
+    )
+
+    if (results !== undefined && results.rows.length > 0) {
+      const client = await getClientWithId(clientId)
+      return results.rows.map(r => createButtonFromRow(r, [client]))
+    }
+
+    if (results.rows.length === 0) {
+      return []
+    }
+  } catch (err) {
+    helpers.logError(err.toString())
+  }
+
+  return null
+}
+
+async function getRecentSessionsWithButtonId(buttonId, pgClient) {
+  try {
+    const results = await helpers.runQuery(
+      'getRecentSessionsWithButtonId',
+      `
+      SELECT s.*
+      FROM sessions AS s
+      LEFT JOIN buttons AS b on s.button_id = b.id
+      WHERE b.id = $1
+      ORDER BY created_at DESC
+      LIMIT 40
+      `,
+      [buttonId],
+      pool,
+      pgClient,
+    )
+
+    if (results !== undefined && results.rows.length > 0) {
+      const allClients = await getClients(pgClient)
+      return results.rows.map(r => createSessionFromRow(r, allClients))
```

Comment:
Ditto here about `createSessionFromRow` expecting an array of Buttons rather than an array of Clients.